imkar/scattering/diffuse.py

```diff
@@ -0,0 +1,178 @@
+
+    sample_area : float
+        Surface area of the measurement sample in square meter. Standard value
+        according to the test sample.
+
+    baseplate_area : float
+        Surface area of the baseplate in square meter. Standard value according
+        to the measurement chamber at IHTA.
+
+    surface_room : float
+        Surface area of the measurement room in square meter. Standard value
+        according to measurement chamber at IHTA.
+
+    volume_room : float
+        Volume of the empty measurement room in cubic meter. Standard value
+        according to measurement chamber at IHTA.
+
+    Returns
+    -------
+    scattering_coefficient : pf.FrequencyData
+        The calculated solution for the diffuse scattering coefficient after
+        International Standard 17497-1:2004(E) and the scattering coefficient
+        of the base plate.
+
+    References
+    ----------
+    ..[#]ISO 17497-1 First edition 2004-05-01
+    """
+    # check inputs
+    if reverberation_time.cshape[-1] != 4:
+        raise ValueError("reverberation_time.cshape has to be (..., 4)")
+
+    if (np.asarray(speed_of_sound, dtype=float).ndim == 1):
+        speed_of_sound = (np.asarray(speed_of_sound, dtype=float)[:, None]).T
+    else:
+        speed_of_sound = (np.asarray(speed_of_sound, dtype=float))
+
+    if (speed_of_sound.shape != reverberation_time.cshape):
+        raise ValueError("speed_of_sound has to be an array containing the \
+                         speed of sound during all 4 measurement conditions. \
+                         Its shape has to be (..., 4) and equal to \
+                         reverberation_time.cshape.")
+
+    if (np.any(np.asarray(speed_of_sound, dtype=float) <= 0)):
+        raise ValueError("speed_of_sound has to be a value bigger than zero")
+
+    if isinstance(air_attenuation, pf.FrequencyData):
+        air_attenuation = np.asarray(air_attenuation.freq, dtype=float)
+    else:
+        air_attenuation = np.asarray(air_attenuation, dtype=float)
+    if air_attenuation.shape != (reverberation_time.cshape +
+                                 (reverberation_time.n_bins, )):
+        raise ValueError("air_attenuation has to be of shape (..., 4,\
+                         number_of_frequencies) and equal the shape of \
+                         reverberation_time.")
+
+    if (np.any(air_attenuation < 0)):
+        raise ValueError("air_attenuation has to be a value bigger or equal \
+                         to zero")
+
+    if calculation_method not in ['ISO', 'Eyring', 'Sabine']:
+        raise ValueError("Use 'ISO', 'Eyring' or 'Sabine' as calculation \
+                         method")
+
+    if (calculation_method == "ISO") & (sample_area != baseplate_area):
+        raise ValueError("For calculation method 'ISO' the baseplate_area and \
+                         sample_area need to be equal.")
+
+    if not isinstance(scale_factor, numbers.Real) or scale_factor <= 0:
```

Comment:
```suggestion
    if not isinstance(scale_factor, (float, int)) or scale_factor <= 0:
```
in this way we dont need numbers package anymore